config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -286,8 +286,8 @@ private static boolean hasAtLeastOneBeanProperty(Class<?> clazz) {
     private static boolean isOptionalProperty(Class beanClass, PropertyDescriptor beanProp) {
         Field field = getField(beanClass, beanProp.getName());
         return field != null
-                ? field.getAnnotationsByType(Optional.class).length > 0
-                        || beanProp.getReadMethod().getAnnotationsByType(Optional.class).length > 0
+                ? beanProp.getReadMethod().getAnnotationsByType(Optional.class).length > 0
```

Comment:
I meant, `beanProp.getReadMethod().getAnnotationsByType(Optional.class).length > 0` could be evaluated at the beginning of method:
```
if (beanProp.getReadMethod().getAnnotationsByType(Optional.class).length > 0)
   return true;
Field field = getField(...);
return field != null && ...